crates/core/src/sql/query_debug_info.rs

```diff
@@ -0,0 +1,32 @@
+use std::sync::Arc;
+
+/// Used to store the source of a SQL query in a way that can be cheaply cloned,
+/// without proliferating lifetimes everywhere.
```

Comment:
QueryDebugInfo needs to get plumbed through more places than ExecutionContext, some of those places it would be adding a lifetime to a non-lifetimed struct, so I didn't want to.